### DISCUSSION TITLE
404 Error while loading up github pages.

### DISCUSSION THREAD

**Discussion by AndreTooGood** [community/community#50933] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I'm not able to access my website for some reason. Everything seems to be correct including the required files. Any help is highly appreciated :) 
andretoogood.github.io/simon-game

**Comment by neutrinomuon** [comment, marked as answer]
Hmm, I think I found the reason. It is being deployed at:

https://andretoogood.github.io/simon-game/Simon%20Game%20Challenge%20Starting%20Files/

Because of the structure of the repo. You probably need to change and include an index.html in your main directory.

I hope this helps you! Please, let me know if it works

**Reply by AndreTooGood** [reply]
I get what you are saying. I should've uploaded the files individually to avoid this issue. However, how do I include an index.html in my main directory?

**Reply by neutrinomuon** [reply]
To clarify, it is important to place your index.html file directly in the main directory of your repository at https://github.com/AndreTooGood/simon-game. This can be achieved either by using the GitHub interface or through traditional commands.

If you have a local repository structure with a directory named "simon-game" and want to upload the index.html file, follow these steps:

1. go to the "simon-game" directory using the command:
<pre>
cd simon-game
</pre>

2.  Place the index.html file in the main directory (simon-game) of the local repository.

3. Use the following commands to add the index.html file, commit the changes, and push it:
<pre>
git add index.html
git commit -m "Added index.html" 
git push ...
</pre>

You can access the URL https://github.com/AndreTooGood/simon-game to check if index.html is placed.
P.S.: Assumed terminal commands above

**Reply by AndreTooGood** [reply]
Appreciate the help. Thank you very much!

**Reply by neutrinomuon** [reply]
Great :)! I'm happy to know it worked

**Reply by Khagesh000** [reply]
My django web pages is showing 404 error
Can anyone fix this error

**Comment by AzizBenIsmail** [comment]
If you are receiving a 404 error when trying to access your GitHub Pages website, it means that the server was not able to locate the requested page. Here are some common reasons why this error may occur and how you can resolve it:

1. Make sure that your website files are in the correct location and named correctly. Your website files should be in a repository named "username.github.io" for personal websites or in a repository named "repositoryname.github.io" for project websites. The main page of your website should be named "index.html".

2. Check if the repository is set up to use GitHub Pages. Go to the repository's settings page and make sure that GitHub Pages is enabled for the repository.

3. Verify that the branch containing your website files is the correct branch. For GitHub Pages, the branch containing your website files should be named "main" or "master".

4. Check if your DNS settings are configured correctly. If you are using a custom domain, make sure that the DNS settings for your domain are configured to point to the correct GitHub Pages IP addresses. You can find the IP addresses in the GitHub Pages settings for your repository.

5. Ensure that there are no issues with the files in your repository. Check the repository for any syntax errors or issues that may cause the website to fail to load.

If you have checked all of the above and still receive a 404 error, try rebuilding your website by pushing your changes to the repository again. If the issue persists, you can contact GitHub support for further assistance.
